crates/core/tedge_watchdog/src/systemd_watchdog.rs

```diff
@@ -243,11 +255,11 @@ async fn monitor_tedge_service(
 async fn get_latest_health_status_message(
     request_timestamp: OffsetDateTime,
     messages: &mut mpsc::UnboundedReceiver<MqttMessage>,
-) -> Result<HealthStatus, WatchdogError> {
+) -> Result<HealthStatusExt, WatchdogError> {
     while let Some(message) = messages.next().await {
         if let Ok(message) = message.payload_str() {
             debug!("Health response received: {message}");
-            if let Ok(health_status) = serde_json::from_str::<HealthStatus>(message) {
+            if let Ok(health_status) = serde_json::from_str::<HealthStatusExt>(message) {
```

Comment:
I feel a bit odd that the watchdog doesn't seem caring if the `status` is `up` or not. As long as `pid` and `time` are included in the health status message, the health status is considered "up"?

Reply:
Yes, if it contains `time` and `pid` it sends systemd-notify notification without checking `status`, and this definitely seems like an overlook to me. I would expect notifications to be sent only when `status` is `up`.

Reply:
> Yes, if it contains `time` and `pid` it sends systemd-notify notification without checking `status`, and this definitely seems like an overlook to me. I would expect notifications to be sent only when `status` is `up`.

I also expect in this way. However, it's out of scope to fix it in this PR?